soc/nxp/imxrt/imxrt118x/soc.c

```diff
@@ -76,6 +76,18 @@ const clock_sys_pll2_config_t sysPll2Config_BOARD_BootClockRUN = {
 	.ssEnable = false,
 };
 
+/* Function Name : BOARD_FlexspiClockSafeConfig
+ * Description   : FLEXSPI clock source safe configuration weak function.
+ *                 Called before clock source configuration.
+ * Note          : Users need override this function to change FLEXSPI clock source to stable
+ *                 source when executing code on FLEXSPI memory(XIP). If XIP, the function
+ *                 should runs in RAM and move the FLEXSPI clock source to a stable clock
+ *                 to avoid instruction/data fetch issue during clock updating.
+ */
+__attribute__((weak)) void BOARD_FlexspiClockSafeConfig(void)
```

Comment:
Are we relocating the `board.c` file in the HAL to RAM? Otherwise, won't this function cause a fault?

Reply:
Agreed. I have rename function to board_flexspi_clock_safe_config. Thank you.